paper_eval.py

```diff
@@ -77,79 +103,102 @@
-    model = Presto.construct(**model_kwargs)
-    best_model_path = None
-model.to(device)
+val_samples_file = f"{task_type}_{test_type}_generalization_test_split_samples.csv"
 
-model_modes = ["Random Forest", "Regression", "CatBoostClassifier"]
+logger.info(f"Preparing train and val splits for {task_type} {test_type} test")
+val_samples_df = pd.read_csv(data_dir / "test_splits" / val_samples_file)
 
-# 1. Using the provided split
-val_samples_df = pd.read_csv(data_dir / val_samples_file)
+if task_type=="croptype":
+    df = WorldCerealBase.map_croptypes(
+        df, 
+        finetune_classes,
+        downstream_classes
+        )
+    df = filter_remove_noncrops(df)
+    
 train_df, test_df = WorldCerealBase.split_df(df, val_sample_ids=val_samples_df.sample_id.tolist())
-full_eval = WorldCerealEval(train_df, test_df, spatial_inference_savedir=model_logging_dir)
-results, finetuned_model = full_eval.finetuning_results(model, sklearn_model_modes=model_modes)
-logger.info(json.dumps(results, indent=2))
 
-model_path = model_logging_dir / Path("models")
-model_path.mkdir(exist_ok=True, parents=True)
-finetuned_model_path = model_path / "finetuned_model_stratified.pt"
-torch.save(finetuned_model.state_dict(), finetuned_model_path)
-
-train_only_samples = pd.read_csv(data_dir / train_only_samples_file).sample_id.tolist()
-# 2. Split according to the countries
-country_eval = WorldCerealEval(
-    *WorldCerealBase.split_df(
-        df,
-        val_countries_iso3=["ESP", "NGA", "LVA", "TZA", "ETH", "ARG"],
-        train_only_samples=train_only_samples,
+full_eval = WorldCerealEval(
+    train_df, test_df, 
+    task_type=task_type,
+    finetune_classes=finetune_classes
     )
-)
-country_results, country_finetuned_model = country_eval.finetuning_results(
-    model, sklearn_model_modes=model_modes
-)
-logger.info(json.dumps(country_results, indent=2))
 
-finetuned_model_path_countries = model_path / "finetuned_model_countries.pt"
-torch.save(country_finetuned_model.state_dict(), finetuned_model_path_countries)
-
-# 3. Split by year
-year_eval = WorldCerealEval(
-    *WorldCerealBase.split_df(df, val_years=[2021], train_only_samples=train_only_samples)
-)
-year_results, year_finetuned_model = year_eval.finetuning_results(
-    model, sklearn_model_modes=model_modes
-)
-logger.info(json.dumps(year_results, indent=2))
+model_path = output_parent_dir / "data" 
+model_path.mkdir(exist_ok=True, parents=True)
+experiment_prefix = f"{presto_model_type}_{compositing_window}_{task_type}_{test_type}_{time_token}"
+finetuned_model_path = model_path / f"{experiment_prefix}.pt"
+results_path = model_logging_dir / f"{experiment_prefix}.csv"
+catboost_model_path = model_logging_dir / f"{experiment_prefix}.cbm"
+
+# check if finetuned model already exists
+logger.info("Checking if the finetuned model exists")
+if os.path.isfile(finetuned_model_path):
+    logger.info("Finetuned model found! Loading...")
```

Comment:
Do we want to do this? I can imagine this introducing lots of unexpected errors since if a finetuned model exists from a previous run, it would automatically affect this run whether or not we want it to

Reply:
Also would this work if there is no finetuned model? I think the script would error out since `finetuned_model` would never be initialized?

Reply:
Also also, how is the model finetuned in this case? I think thats pretty important so it would be good to capture that in this script

Reply:
I still feel the need for being able to upload the finetuned model, particularly if I want to run downstream classifiers only and collect metrics. in particular, [this piece](https://github.com/WorldCereal/presto-worldcereal/blob/5f5d296962fb158f6f4c9f84e74e0deff2efcde7/paper_eval.py#L173-L200) only checks if the model exists, collects metrics for the uploaded model along with spatial plots and runs sklearn models. 
I didn't succeed in implementing the upload in presto.py, so now this piece wouldn't run anyway. for now, I
just put a placeholder here https://github.com/WorldCereal/presto-worldcereal/commit/532ec67b5a81b0845f05534864550de073957fb3